### PR TITLE
DEP: integrate: Deprecate `initial` values other than None or 0 in cumulative_trapezoid

### LINKED ISSUE #18588: DOC: scipy.integrate.cumulative_trapezoid initial clarification
### Issue with current documentation:

Currently it is stated that the `initial` argument of `scipy.integrate.cumulative_trapezoid` is inserted as the resulting array of the cumulative integration.

However, I would have expected a cumulative function to add the `initial` to *all* values (since it is part of the cumulative function).

However, this is not clear from the documentation.

### Idea or request for content:

I can see that this may really break things, to I would suggest to add a small example clarifying the `initial` argument:

```python
from scipy.integrate import cumulative_trapezoid

print(cumulative_trapezoid([0, 1], dx=1, initial=2))
```
which yields `[2.   0.5]`

To get the initial in a cumulative fashion one should do:
```python
print(cumulative_trapezoid([0, 1], dx=1, initial=0)+2)
```
which yields `[2.   2.5]`

### Additional context (e.g. screenshots, GIFs)

_No response_

### PR DESCRIPTION
**Pull request by nprav**
#### Reference issue

Related to [#125](https://github.com/scipy/scipy/pull/125`)

#### What does this implement/fix?

The `initial` kw in `integrate.cumulative_trapezoid` was only intended as a way to make the output array have the same length as the input array. Ie: if initial=None, len(output) = len(input) - 1. If initial=0, len(output) = len(input), and the first value is 0.
But, `initial` can currently be any other scalar, and `cumulative_trapezoid` will prepend the value to the result. This is confusing because `initial` may be assumed to be an initial condition to the integration which is not true.

To remove the confusion this PR updates `cumulative_trapezoid` to raise a ValueError if `initial` is not None or 0.

#### Additional information

This issue has been discussed in #18151 and a Scipy mailing list [thread](https://mail.python.org/archives/list/scipy-dev@python.org/thread/5UYCLPVODQJRMMXCUQA5DNPBOIBKAJPK/).

### CODE REVIEW

**Review by MatteoRaso — CHANGES_REQUESTED** [review]
Thanks for the PR. It looks great, just one tiny thing that I recommend changing to fix the test failures.

File: scipy/integrate/_quadrature.py [hunk 1]
```diff
@@ -472,8 +475,8 @@ def cumulative_trapezoid(y, x=None, dx=1.0, axis=-1, initial=None):
     res = np.cumsum(d * (y[slice1] + y[slice2]) / 2.0, axis=axis)
 
     if initial is not None:
-        if not np.isscalar(initial):
-            raise ValueError("`initial` parameter should be a scalar.")
+        if initial is not 0:
```
Comment (MatteoRaso) [inline scipy/integrate/_quadrature.py:478]:
```suggestion
        if initial != 0:
```
This is what's causing the test failures.

**Review by MatteoRaso — COMMENTED** [review]
Most of the test failures seem to be unrelated, but there's actually a linting error I missed when reviewing.

**Review by MatteoRaso — APPROVED** [review]
Thanks for implementing my suggestions. LGTM.

**Comment by j-bowhay** [comment]
I think it would be safer to go through a deprecation cycle to make this change

**Comment by nprav** [comment]
Thanks @j-bowhay , that makes sense to me. How would we go about doing that? Should I make a new PR with just the Deprecation warning, and then add the changes in this PR in a later release?

**Comment by j-bowhay** [comment]
I think we can stay on the same pr, effectively all we need to do is change where you raise the exception to a deprecation warning and make a slight tweak to the docs and test

File: scipy/integrate/tests/test_quadrature.py [hunk 2]
```diff
@@ -322,18 +322,28 @@ def test_x_none(self):
...
+        y_expected = [0, -4.5, -6., -4.5, 0.]
         assert_allclose(y_int, y_expected)
 
+    @pytest.mark.parametrize(
+        "initial", [1, 0.5]
+    )
+    def test_initial_warning(self, initial):
+        """If initial is not None or 0, a ValueError is raised."""
+        y = np.linspace(0, 10, num=10)
+        with pytest.deprecated_call():
```
Comment (j-bowhay) [inline scipy/integrate/tests/test_quadrature.py:343]:
```suggestion
        with pytest.deprecated_call(match="`initial`"):
```

**Review by j-bowhay — COMMENTED** [review]
Thanks @nprav, one final nit

File: scipy/integrate/tests/test_quadrature.py (hunk 2 above)
Comment (nprav) [reply scipy/integrate/tests/test_quadrature.py]: Thanks @j-bowhay , added in [aba400b](https://github.com/scipy/scipy/pull/18219/commits/aba400b702b2af0fec52a6607e6c547688cab6cd)

File: scipy/integrate/_quadrature.py [hunk 3]
```diff
@@ -472,6 +477,13 @@ def cumulative_trapezoid(y, x=None, dx=1.0, axis=-1, initial=None):
     res = np.cumsum(d * (y[slice1] + y[slice2]) / 2.0, axis=axis)
 
     if initial is not None:
+        if initial != 0:
+            warnings.warn(
+                "The option for values for `initial` other than None or 0 is "
+                "deprecated as of SciPy 1.12.0 and will be removed in SciPy "
+                "1.14.0.",
```
Comment (j-bowhay) [inline scipy/integrate/_quadrature.py:484]:
```suggestion
                "deprecated as of SciPy 1.12.0 and will raise a value error in"
                " SciPy 1.14.0.",
```

**Review by j-bowhay — COMMENTED** [review]
Thanks @nprav  for tackling this, good to get rid of another weird behaviour. Congratulations on what looks like your first contribution to SciPy!!

**Comment by nprav** [comment]
Thanks for the reviews @j-bowhay !

**Pull request merged**